packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2840,18 +2842,15 @@ class _WidgetInspectorState extends State<WidgetInspector>
       _inspectAt(_lastPointerLocation!);
       WidgetInspectorService.instance._sendInspectEvent(selection.current);
     }
-    setState(() {
-      // Only exit select mode if there is a button to return to select mode.
-      if (widget.selectButtonBuilder != null) {
-        isSelectMode = false;
-      }
-    });
+
+    // Only exit select mode if there is a button to return to select mode.
+    if (widget.selectButtonBuilder != null) {
+      isSelectMode = false;
```

Comment:
this line will be clearer written as 
isSelectMode.value = true;

as that doesn't make it look like you are missing a setState call.